live-build/misc/ansible-roles/appliance-build.minimal-common/tasks/main.yml

```diff
@@ -323,3 +323,37 @@
   with_items:
     - { key: 'kernel.core_pattern', value: '/var/crash/core.%e.%p.%t' }
     - { key: 'vm.overcommit_memory', value: '1' }
+
+#
+# Add the apt repos that have the debug versions of packages, so that we
+# can download a debug kernel to use with crash.
+#
+- apt:
+    name: 'ubuntu-dbgsym-keyring'
+- apt_repository:
+    repo: "{{ item }}"
+  with_items:
+    - deb http://ddebs.ubuntu.com bionic main restricted universe multiverse
+    - deb http://ddebs.ubuntu.com bionic-updates main restricted universe multiverse
+    - deb http://ddebs.ubuntu.com bionic-proposed main restricted universe multiverse
+
+#
+# Assume that the newest kernel is the one we are going to boot, and
+# fetch the symbols for that one.
+#
+- shell: ls /boot/vmlinuz-* | sed 's|/boot/vmlinuz-||' | sort -V | tail -n1
+  register: kernel_version
+
+- apt:
+    name: 'linux-image-{{ kernel_version.stdout }}-dbgsym'
+
+#
+# Increase the amount of memory reserved for the crash kernel.
+# Empirically, it seems we need about 512M to get it to boot. If the
+# system has less than 1G, disable crash dumps rather reserving half of
+# memory for the crash kernel.
+#
+- lineinfile:
+    path: /etc/default/grub.d/kdump-tools.cfg
+    regexp: '^GRUB_CMDLINE_LINUX_DEFAULT='
+    line: 'GRUB_CMDLINE_LINUX_DEFAULT="$GRUB_CMDLINE_LINUX_DEFAULT crashkernel=1024M-:512M"'
```

Comment:
Why doesn’t the default values work? What are the defaults?

Reply:
The default is 128M, but with that amount of memory the kernel panics before it even starts the first userspace process because it runs out of memory. This doesn't appear to be related to any configuration that is specific to Delphix, because it happens on a standard install of Ubuntu too. I haven't looked too closely at where exactly all the memory is going. I'm planning on filing an upstream bug to see whether there is any way to reduce the amount of memory needed, or at least to update the default value.

Reply:
Thanks. This looks good, was just curious.

Reply:
I know you've been talking to the maintainers of this package, have you talked to them about increasing the default memory when using zfs too? It would be nice to have the package just handle this correctly.

Reply:
@grwilson I actually don't think this is specific to ZFS, the amount of memory needs to be increased even when using an ext4 root filesystem. I am planning on following up with the package maintainers though.

Reply:
Great thanks!